stdlib/REPL/src/REPL.jl

```diff
@@ -44,7 +44,13 @@ function UndefVarError_hint(io::IO, ex::UndefVarError)
                 if C_NULL == owner
                     # No global of this name exists in this module.
                     # This is the common case, so do not print that information.
-                    print(io, "\nSuggestion: check for spelling errors or missing imports.")
+                    # It could be the binding was exported by two modules, which we can detect
+                    # by the `usingfailed` flag in the binding:
+                    if isdefined(bnd, :flags) && Bool(bnd.flags >> 4 & 1) # magic location of the `usingfailed` flag
+                        print(io, "\nHint: It looks like this name was exported by two different modules, resulting in ambiguity. Try explicitly importing it, or qualifying the name with the module it should come from.")
```

Comment:
This suggestion addresses the following:
1. Reduces the line length by splitting up the string, since Base sets its line length to 92 characters.
2. Notes that the issue occurs specifically because the bindings with the same name refer to different things, since this won't occur when two modules export the same name and it refers to the same thing.
3. Clarifies that the issue may not be limited to two modules.
```suggestion
                        print(io, "\nHint: It looks like two or more modules export different ",
                              "bindings with this name, resulting in ambiguity. Try explicitly ",
                              "importing it from a particular module, or qualifying the name ",
                              "with the module it should come from.")
```

Reply:
looks good to me!